force-app/main/default/classes/RD2_QueueableService.cls

```diff
@@ -94,8 +94,9 @@ public class RD2_QueueableService {
         }
 
         EvaluateInstallmentOpportunities service = new EvaluateInstallmentOpportunities(toProcess, rdIdsWhereScheduleChanged, oldRds);
+        RD2_Settings settings = RD2_Settings.getInstance();
 
-        if (Limits.getQueueableJobs() < Limits.getLimitQueueableJobs()) {
+        if (Limits.getQueueableJobs() < Limits.getLimitQueueableJobs() && !settings.isGiftEntryMode) {
```

Comment:
The logic on checking if it's in gift Entry mode should be moved to `RD2_RecurringDonations_TDTM` [line 403](https://github.com/SalesforceFoundation/NPSP/blob/db131092fef2093f645c2b394d9192903e469c49/force-app/main/default/classes/RD2_RecurringDonations_TDTM.cls#L403). `RD2_QueuableService` should only be used to determine if a queuable can be launched and should not have any business logic in there. RD2 TDTM has a method to determine if RDs should run in async mode.